astroquery/alma/core.py

```diff
@@ -341,8 +363,8 @@ def query_async(self, payload, cache=None, public=True, science=True,
             return payload
 
         query = _gen_sql(payload)
-        result = self.query_tap(query, **kwargs)
-        if result:
+        result = self.query_tap(query, maxrec=payload.get('maxrec', None))
+        if result is not None:
```

Comment:
should the alternative be to raise some sort of exception rather than `return None`, as it presently is?

Reply:
True. It should never happen so I've changed it to raise an exception now.